### PR TITLE
Keep using default navigation animation direction with RTL languages

### PR DESCRIPTION
**Pull request by ManueGE**
If the user is using an RTL (right to left) language, the navigation controller is expected to use the inverse transition directions by default: `.right` for pushing and `.left` for popping. 

Currently, it is not working, and the navigation controllers behave as it does in LTR languages. 

This commit fixes this issue.

### CODE REVIEW

**Comment by SD10** [comment]
Hey @ManueGE,

Thanks for this PR 👍 I'll have to dig a bit deeper regarding this one. My first thought is -- are you able to easily achieve this effect at the client level?

I don't really like the idea of changing the behavior of `.left` and `.right` based on the user's language. I think these directions should be taken quite literally -- as they are with the Apple APIs. 

I think it's better to use the terms `.leading` and `.trailing` for directions that change based on a users language

**Comment by ManueGE** [comment]
I'm actually not changing the behavior of `.left` or `.right`, I am just choosing one of them for the `.auto` animation based on the user language. 

Adding a `.leading` and `.trailing` direction maybe is a little more semantic, so we can go this way too.

I can fix it at client level easily, but anyway I am using my own fork of Hero in my app because it has another change I needed (#513), so it is not blocking me at all. 

Thanks,,

**Comment by ManueGE** [comment]
I added `.leading` and `.trailing`, but instead of adding a new case in the enum I made them computed static vars, so they can be used as cases.

Also added `.leadingToTrailing` and `.trailingToLeading` in `CascadeDirection`.

What do you think?

**Comment by SD10** [comment]
@ManueGE This seems both reasonable and flexible to me, thanks 👍 Going to see if anyone else has some input on this issue so that I'm not dictating a decision

**Comment by ManueGE** [comment]
Any new with this PR?

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift [hunk 1]
```diff
@@ -249,9 +259,9 @@ class DefaultAnimationPreprocessor: BasePreprocessor {
       if animators.contains(where: { $0.canAnimate(view: toView, appearing: true) || $0.canAnimate(view: fromView, appearing: false) }) {
         defaultAnimation = .none
       } else if inNavigationController {
-        defaultAnimation = presenting ? .push(direction:.left) : .pull(direction:.right)
+        defaultAnimation = presenting ? .push(direction:.leading) : .pull(direction:.trailing)
```
Comment (SD10) [inline Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: @ManueGE The only thing that concerns me is now we're changing the behavior of this animation. I agree that it's nice to change things based on the `userInterfaceLayoutDirection` -- but a Hero user could be relying on this not being changed

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift [hunk 2]
```diff
@@ -249,9 +259,9 @@ class DefaultAnimationPreprocessor: BasePreprocessor {
...
       } else if inTabBarController {
-        defaultAnimation = presenting ? .slide(direction:.left) : .slide(direction:.right)
+        defaultAnimation = presenting ? .push(direction:.leading) : .pull(direction:.trailing)
```
Comment (SD10) [inline Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: Why did we change from `.slide` to `.push` here?

**Comment by SD10** [comment]
@ManueGE Yeah, I was able to review this a bit and left you some comments above. Sorry I've been swamped at work for months now 😞 If you need me to look at something the best way is to ping me on GitHub or DM me on Twitter 😓

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift (hunk 2 above)
Comment (ManueGE) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: oops, you are right. My bad, will fix it.

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift (hunk 1 above)
Comment (ManueGE) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: You are totally right, I understand that some users could be relying on this even if maybe it's not the "right" behavior. I saw it as a bug fix, but I understand your point.

Comment (SD10) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: Can we make this configurable so that users can opt-in to the old behavior?

Comment (ManueGE) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: Let me think about it a little bit. Will try to figure out something tomorrow

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift (hunk 2 above)
Comment (ManueGE) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: Fixed

File: Sources/Preprocessors/DefaultAnimationPreprocessor.swift (hunk 1 above)
Comment (ManueGE) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: I kept the "left to right" default behavior, but "user interface" and "right to left" behaviors can be set using the `defaultTransitionDirectionStrategy` property of `HeroTransition`.

Comment (SD10) [reply Sources/Preprocessors/DefaultAnimationPreprocessor.swift]: Thanks @ManueGE. I skimmed this earlier this week. This looks totally reasonable 👍 Will review tomorrow and hopefully merge + release!

**Pull request merged**